### PR TITLE
adding PageLayout and examples

### PR DESCRIPTION
**Pull request by FoamyGuy**
A new type of layout that allows you to organize your UI into pages that you can change between using name, index, or prev/next functions.

### CODE REVIEW

File: adafruit_displayio_layout/layouts/page_layout.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        KeyError if the page was not found in the PageLayout.
+
+        :param string page_name: the name of the page to lookup
+        :param int page_index: the index of the page to lookup
+        :return: the displayio content object at those coordinates
+        """
+
+        self._check_args(page_name, page_index)
+
+        if page_index:
```
Comment (rsbohn) [inline adafruit_displayio_layout/layouts/page_layout.py]: This should be `if page_index is not None:`

File: adafruit_displayio_layout/layouts/page_layout.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+            return self._page_content_list[page_index]
+
+        if page_name:
```
Comment (rsbohn) [inline adafruit_displayio_layout/layouts/page_layout.py]: This should be `if page_name is not None:`

**Review by rsbohn — CHANGES_REQUESTED** [review]

**Comment by rsbohn** [comment]
Consider adding a property to retrieve the current page content:

    showing_page_content -> Displayable

Make it easy for the user to retrieve the object they added using `add_content()` without any wrappers.

The user shouldn't need to know about `_page_group` and `sub_view_obj`, these are implementation details.

File: adafruit_displayio_layout/layouts/page_layout.py (hunk 1 above)
Comment (FoamyGuy) [reply adafruit_displayio_layout/layouts/page_layout.py]: Good catch thank you. Fixed with latest commit

File: adafruit_displayio_layout/layouts/page_layout.py (hunk 2 above)
Comment (FoamyGuy) [reply adafruit_displayio_layout/layouts/page_layout.py]: Good catch thank you. Fixed with latest commit

**Comment by FoamyGuy** [comment]
> Consider adding a property to retrieve the current page content:
> 
> ```
> showing_page_content -> Displayable
> ```
> 
> Make it easy for the user to retrieve the object they added using `add_content()` without any wrappers.
> 
> The user shouldn't need to know about `_page_group` and `sub_view_obj`, these are implementation details.

Good idea, I like that functionality.  Thanks for trying this out and suggesting improvements. I'll add another commit with this new property

**Comment by FoamyGuy** [comment]
Also I like the idea of a `Displayable` typing protocol (I think that is the right term). Lots of functions can accept this type of argument. Originally TileGrid and Group were the only that existed but now the vectorio shapes exist and are similar. Having this type defined somewhere will allow us to more easily update it if we end up adding any more types that can be shown directly like this.

**Comment by FoamyGuy** [comment]
Latest commit adds the `showing_page_content` property as well as an illustration of it's usage in the advanced example script

**Comment by FoamyGuy** [comment]
Not sure of the cause of CI failure. It passes locally for me:
image

Will look into this a bit later on today.

**Comment by FoamyGuy** [comment]
I think this is the root caused of the CI failure. https://github.com/psf/black/issues/2964 Looks like there are a few options to get past it but it might be something we'll need to decided on across all of the libraries if I am understanding correctly.

**Comment by FoamyGuy** [comment]
Indeed the Black / Click version issue is what was causing CI to fail. There was a commit yesterday to change the version used and I just re-ran the jobs and now it's passing :tada:

**Comment by aioue** [comment]
**LGTM.**

Tested code on a Titano successfully.

Test process:
- Loaded `page_layout.py` onto a Titano
- Ran the 2 test files using adapted import line `from page_layout import PageLayout` (hobbyist python user)

Test results:
- `simpletest` PASS. Rendered both pages successfully and switches between them.
- `advancedtest` PASS. Rendered all three pages successfully and modifies the second page as expected.

Notes:
- Feels faster to load the code and render the text and shapes than other Learning System templates 🚀  Nice work FoamyGuy 👏

**Pull request merged**